services/director-v2/src/simcore_service_director_v2/modules/db/repositories/internet_to_groups.py

```diff
@@ -0,0 +1,22 @@
+import sqlalchemy as sa
+from models_library.users import UserID
+
+from ..tables import internet_to_groups, user_to_groups
+from ._base import BaseRepository
+
+
+class InternetToGroupsRepository(BaseRepository):
+    async def has_access(self, user_id: UserID) -> bool:
+        async with self.db_engine.acquire() as conn:
+            # checks if one of the groups which the user is part of has internet access
+            select_stmt = sa.select([internet_to_groups.c.has_access]).select_from(
+                user_to_groups.join(
+                    internet_to_groups,
+                    (internet_to_groups.c.group_id == user_to_groups.c.gid)
+                    & (user_to_groups.c.uid == user_id)
```

Comment:
Q: so the idea is that you get the permissions of the group with more privileges? i.e. if you belong to three groups, you get the permissions with the one with most privileges, right?